geode-management/src/main/java/org/apache/geode/management/internal/ClusterManagementClient.java

```diff
@@ -0,0 +1,107 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more contributor license
+ * agreements. See the NOTICE file distributed with this work for additional information regarding
+ * copyright ownership. The ASF licenses this file to You under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance with the License. You may obtain a
+ * copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software distributed under the License
+ * is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express
+ * or implied. See the License for the specific language governing permissions and limitations under
+ * the License.
+ */
+
+package org.apache.geode.management.internal;
+
+import org.springframework.http.ResponseEntity;
+import org.springframework.http.client.ClientHttpRequestFactory;
+import org.springframework.web.client.ResponseErrorHandler;
+import org.springframework.web.client.RestTemplate;
+import org.springframework.web.util.DefaultUriTemplateHandler;
+import sun.reflect.generics.reflectiveObjects.NotImplementedException;
+
+import org.apache.geode.annotations.Experimental;
+import org.apache.geode.cache.configuration.CacheElement;
+import org.apache.geode.management.api.ClusterManagementResult;
+import org.apache.geode.management.api.ClusterManagementService;
+import org.apache.geode.management.api.RestfulEndpoint;
+import org.apache.geode.management.client.ClusterManagementServiceProvider;
+import org.apache.geode.management.client.RestTemplateResponseErrorHandler;
+
+/**
+ * Implementation of {@link ClusterManagementService} interface which represents the cluster
+ * management service as used by a Java client.
+ * <p/>
+ * In order to manipulate Geode components (Regions, etc.) clients can construct instances of {@link
+ * CacheElement}s and call the corresponding {@link ClusterManagementClient#create(CacheElement,
+ * String)}, {@link ClusterManagementClient#delete(CacheElement, String)} or {@link
+ * ClusterManagementClient#update(CacheElement, String)} method. The returned {@link
+ * ClusterManagementResult} will contain all necessary information about the outcome of the call.
+ * This will include the result of persisting the config as part of the cluster configuration as
+ * well as creating the actual component in the cluster.
+ * <p/>
+ * All create calls are idempotent and will not return an error if the desired component already
+ * exists.
+ */
+@Experimental
+public class ClusterManagementClient implements ClusterManagementService {
```

Comment:
Isn't this the same as `GeodeClientClusterManagementService` and `ServerClusterManagementService`?